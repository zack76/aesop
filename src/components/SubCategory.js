import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import {Accordion, Card, Button, Col, Row} from "react-bootstrap";
import Item from "./Item"


const SubCategory = ({ subcategories }) => {
    return subcategories.map((subcategory, index) =>
        (
            <Accordion key={subcategory.name + index} defaultActiveKey="0">
                <Card>
                    <Card.Header>
                        <Accordion.Toggle as={Button} variant="link"
                                          eventKey={'subcategory' + subcategory.name + index}>
                            {subcategory.name}
                        </Accordion.Toggle>
                    </Card.Header>
                    <Accordion.Collapse eventKey={'subcategory' + subcategory.name + index}>
                        <Card.Body>
                            {subcategory.items[0].price ?
                                <Col xs={12} key={subcategory.name + index}>
                                    <Row>
                                        <Item items={subcategory.items}/>
                                    </Row>
                                </Col>
                                :
                                <SubCategory key={'subcategory' + subcategory.name + index}
                                subcategories={subcategory.items}/>
                            }
                        </Card.Body>
                    </Accordion.Collapse>
                </Card>
            </Accordion>
        ))
}

export default SubCategory
