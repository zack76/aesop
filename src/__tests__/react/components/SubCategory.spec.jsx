import React from 'react';
import { mount, shallow } from 'enzyme';
import SubCategory from '../../../components/SubCategory';
import { Accordion, Button } from 'react-bootstrap';

jest.mock('../../../components/Item', () => 'item-list')
describe('SubCategory', () => {
    const props = {
        subcategories: [
            { name: 'toy', items: [
                    { name: 1, price: 2, variants: [ {a:1, b: 2}]},
                    { name: 2, price: 2, variants: [ {a:1, b: 2}]},
                    { name: 3, price: 2, variants: [ {a:1, b: 2}]}
                ] },
        ]
    };

    it('Should render without error', () => {
        const wrapper = mount(<SubCategory {...props} />);
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find(Accordion).length).toEqual(props.subcategories.length);
        expect(
            wrapper
                .find(Accordion.Toggle)
                .first()
                .text()
        ).toBe('toy');
        expect(
            wrapper
                .find('item-list')
                .length
        ).toEqual(1);
    })

    it('Should render item with no price without error', () => {
        const props = {
            subcategories: [
                { name: 'toy', items: [
                        { name: 'toy', items: [
                                { name: 1, price: 2, variants: [ {a:1, b: 2}]},
                                { name: 2, price: 2, variants: [ {a:1, b: 2}]},
                                { name: 3, price: 2, variants: [ {a:1, b: 2}]}
                            ] }
                    ] },
            ]
        };
        const wrapper = mount(<SubCategory {...props} />);
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find(Accordion).length).toEqual(2);
    })

    it('Should have Accordion.Toggle with "as" is Button', () => {
        const wrapper = shallow(<SubCategory {...props} />);
        expect(
            wrapper
                .find(Accordion.Toggle)
                .first()
                .prop('as')
        ).toBe(Button);
        expect(wrapper).toMatchSnapshot();
    });
});
