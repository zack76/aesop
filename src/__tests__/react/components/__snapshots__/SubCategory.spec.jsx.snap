// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Should render without error 1`] = `
<ProductList
  products={
    Array [
      Object {
        "items": Array [
          "a",
          "b",
          "c",
        ],
        "name": "toy",
      },
      Object {
        "items": Array [
          "one",
          "two",
          "three",
        ],
        "name": "paper",
      },
      Object {
        "items": Array [
          "apple",
          "pear",
          "orange",
        ],
        "name": "food",
      },
    ]
  }
>
  <ForwardRef
    defaultActiveKey="0"
    key="0"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="category0"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    toy
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="category0"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <sub-category
                        subcategories={
                          Array [
                            "a",
                            "b",
                            "c",
                          ]
                        }
                      />
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
  <ForwardRef
    defaultActiveKey="0"
    key="1"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="category1"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    paper
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="category1"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <sub-category
                        subcategories={
                          Array [
                            "one",
                            "two",
                            "three",
                          ]
                        }
                      />
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
  <ForwardRef
    defaultActiveKey="0"
    key="2"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="category2"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    food
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="category2"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <sub-category
                        subcategories={
                          Array [
                            "apple",
                            "pear",
                            "orange",
                          ]
                        }
                      />
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
</ProductList>
`;

exports[`SubCategory Should render without error 1`] = `
<SubCategory
  subcategories={
    Array [
      Object {
        "items": Array [
          Object {
            "name": 1,
            "price": 2,
            "variants": Array [
              Object {
                "a": 1,
                "b": 2,
              },
            ],
          },
          Object {
            "name": 2,
            "price": 2,
            "variants": Array [
              Object {
                "a": 1,
                "b": 2,
              },
            ],
          },
          Object {
            "name": 3,
            "price": 2,
            "variants": Array [
              Object {
                "a": 1,
                "b": 2,
              },
            ],
          },
        ],
        "name": "toy",
      },
    ]
  }
>
  <ForwardRef
    defaultActiveKey="0"
    key="toy0"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="subcategorytoy0"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    toy
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="subcategorytoy0"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <Col
                        key="toy0"
                        xs={12}
                      >
                        <div
                          className="col-12"
                        >
                          <Row
                            noGutters={false}
                          >
                            <div
                              className="row"
                            >
                              <item-list
                                items={
                                  Array [
                                    Object {
                                      "name": 1,
                                      "price": 2,
                                      "variants": Array [
                                        Object {
                                          "a": 1,
                                          "b": 2,
                                        },
                                      ],
                                    },
                                    Object {
                                      "name": 2,
                                      "price": 2,
                                      "variants": Array [
                                        Object {
                                          "a": 1,
                                          "b": 2,
                                        },
                                      ],
                                    },
                                    Object {
                                      "name": 3,
                                      "price": 2,
                                      "variants": Array [
                                        Object {
                                          "a": 1,
                                          "b": 2,
                                        },
                                      ],
                                    },
                                  ]
                                }
                              />
                            </div>
                          </Row>
                        </div>
                      </Col>
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
</SubCategory>
`;
